### PR TITLE
Update setup to declarative metadata

### LINKED ISSUE #1864: Package includes test.geometry module
The package distribution includes `test.geometry`.

setup.py needs:

```diff,python
-        packages=find_packages(exclude=('docs', 'test', 'examples')),
+        packages=find_packages(exclude=('docs', 'test', 'test.*', 'examples')),
```

(Might as well go wildcard.)

### PR DESCRIPTION
**Pull request by johnnv1**
#### Changes
Move the kornia setup to declarative metadata (`setup.cfg`). With this, we have a simple `setup.py` and all information are at `setup.cfg`.  Also add a pre-commit hook to ensure the config format (https://github.com/asottile/setup-cfg-fmt)

I remove the "kornia/_version.py" module, but I don't think this should be a problem. The `__version__` still there, and the info will be load with  `importlib_metadata`.

I move the pytest markers to `setup.cfg` and remove `pytest.ini` file. Also, remove the duplicated `isort` config from `setup.cfg` (the isort use configs from `pyproject.toml` by default)

Fixes #1864 

----

We also can remove the requirements files and add as extra:
```diff
+[options.extras_require]
+dev =
+    flake8==5.0.4
+    isort
+    numpy
+    opencv-python
+    pre-commit>=2.0
+    pydocstyle
+    pytest==7.1.3
+    pytest-cov==3.0.0
+    pytest-mypy==0.9.1
+    scipy
+    kornia-rs==0.0.5;python_version>='3.7'
+x =
+    accelerate==0.12.0
+    dataclasses;python_version<'3.7'
+docs =
+    torch
+    torchvision
+    opencv-python
+    matplotlib
+    sphinx>=4.0
+    sphinx_rtd_theme>0.5
+    sphinxcontrib-bibtex
+    sphinxcontrib-youtube
+    sphinx-autodoc-typehints>=1.0
+    sphinx-copybutton>=0.3
+    sphinx-autodoc-defaultargs
+    sphinxcontrib-gtagjs
+    PyYAML>=5.1,<6.1.0
+    furo
+    sphinx-design
```

and use with:
```bash
$ pip install -e ".[dev]" # for dev requirements

$ pip install -e ".[x]" # for x requirements

$ pip install -e ".[docs]" # for docs requirements
```

### CODE REVIEW

File: kornia/__init__.py [hunk 1]
```diff
@@ -14,3 +11,13 @@
...
+
+# Version variable
+import sys
+
+if sys.version_info >= (3, 8):  # pragma: >=3.8 cover
+    import importlib.metadata as importlib_metadata
+else:  # pragma: <3.8 cover
+    import importlib_metadata
+
+__version__ = importlib_metadata.version('kornia')
```
Comment (edgarriba) [inline kornia/__init__.py:23]: for this one, i recently discovered `version = attr:` in a separate project i'm working on
[version = attr:](https://github.com/farm-ng/amiga-dev-kit/blob/main/brain/setup.cfg#L3)

less code but i kinda feel that having in setup.cfg seems more direct and find to easy. Any opinions ?

Comment (johnnv1) [reply kornia/__init__.py:23]: Personally, I prefer it directly in setup.cfg because that's where you would expect it to be

Comment (edgarriba) [reply kornia/__init__.py:23]: sounds good to me

**Comment by edgarriba** [comment]
@johnnv1 testing workflow apparently got in a bad state. All good i believe now

File: setup.cfg [hunk 2]
```diff
@@ -10,12 +67,10 @@ norecursedirs =
 addopts =
     --strict
     --color=yes
-
-[bdist_wheel]
-universal=1
-
-[metadata]
-license_file = LICENSE
+markers =
```
Comment (edgarriba) [inline setup.cfg:70]: I believe we can remove this markers, they are not used anymore. Possibly in a separated pr

File: setup.cfg [hunk 3]
```diff
@@ -48,15 +110,3 @@ exclude_lines =
     raise
     if 0:
     if __name__ == .__main__.:
-
-[isort]
```
Comment (edgarriba) [inline setup.cfg:52]: Why do you remove this ?

**Review by edgarriba — COMMENTED** [review]
Seems that pytest is not installed correctly in the ci. Maybe it’s related to the requirements, I’m okay with your suggestion to add in the setup the requirements. Can you make combinations?

File: setup.cfg (hunk 3 above)
Comment (johnnv1) [reply setup.cfg:52]: I think this config is duplicated, the isort use configs from `pyproject.toml` by default

File: setup.cfg (hunk 2 above)
Comment (johnnv1) [reply setup.cfg:70]: will be better in a separated PR

**Comment by johnnv1** [comment]
> Seems that pytest is not installed correctly in the ci. Maybe it’s related to the requirements, I’m okay with your suggestion to add in the setup the requirements. Can you make combinations?

Yeah, I saw just now that CI is installing the dependencies with `pip install .[all]`, moving the dependencies to “setup.cfg”, can we also remove the files at “requirements/“?

**Comment by johnnv1** [comment]
Also, in ".github/workflows/format.yml":
- The `linter` job, seems unnecessary since it is already provided for in the pre-commit hooks
- The pre-commit action is deprecated, and the pre-commit.ci (already configured at the repo) do the same job

**Comment by johnnv1** [comment]
I change the CI to install the specifics extras, can you rerun the CI @edgarriba ? 😅

**Comment by edgarriba** [comment]
This needs to be updated too https://github.com/kornia/kornia/blob/master/.github/workflows/tests_lts.yml

**Comment by johnnv1** [comment]
I merged this new CI actions and update it

**Comment by johnnv1** [comment]
there was one `]` extra, now all CI should be corrected 😅

File: setup.cfg (hunk 2 above)
Comment (ducha-aiki) [reply setup.cfg:103]: @edgarriba wait, we do use markers, not? How we skip (some) of jit tests then? I also use them when run tests locally

**Comment by YanivHollander** [comment]
Hi @johnnv1,

The latest update to `kornia/kornia/__init__.py` breaks Kornia when importing the library in a Colab notebook.

The line

```
...
__version__ = importlib_metadata.version('kornia') 
...
```

raises the exception: `PackageNotFoundError: No package metadata was found for kornia`.
From within a running Docker Kornia import works fine, but in Colab Docker won't work, so I have to import the module for dev purposes NOT from within a Docker. Any ideas? Also please note that currently Colab runs Python3.7...

cc @edgarriba

**Comment by johnnv1** [comment]
Maybe some specific problem at the docker init script?

I try this in a Colab Notebook and works fine to me:

cell#1
```python
!pip install git+https://github.com/kornia/kornia.git -q

import kornia
print(kornia.__version__)
```
```output
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
    Preparing wheel metadata ... done
0.6.8.dev0
```

cell#2
```python
import importlib_metadata

print(importlib_metadata.version('kornia'))
```
```output
0.6.8.dev0
```

**Comment by YanivHollander** [comment]
@johnnv1,

I guess what I'm trying to do that is different than what you showed is that I import Kornia in a Colab notebook directly from a clone of it on my Google Drive. I'm neither `pip install` it nor import it from within this project's image running container. 

As a workaround I'm hardcoding a value for `__version__`, so this line does not throw the `PackageNotFoundError` exception. It's not critical, but it would be nice if it worked the same as before your changes.

**Comment by johnnv1** [comment]
I get it.

To get the `__version__` from directly the source instead of the package installed info will be an ugly thing to have, I think will be something like:
```python
from setuptools.config import read_configuration 
__version__ = read_configuration("setup.cfg")["metadata"]["version"]
```

personally, I prefer just test it before:
```python
if 'kornia' in sys.modules:
    __version__ = importlib_metadata.version('kornia') 
```

@edgarriba any other ideia here?

**Comment by edgarriba** [comment]
the second one seems more clean

**Pull request merged**